### DISCUSSION TITLE
Flux with ACR seamless integration

### DISCUSSION THREAD

**Discussion by michalgozdowski** [fluxcd/flux2#3553] [Q&A]
Hi, 

I am running Flux as an extensions in AKS cluster which has ACR integration enabled and configured using kubelet identity. Following the information found here: https://fluxcd.io/flux/use-cases/azure/#image-updates-with-azure-container-registry this should be enough to automate image updates, but still `image-reflector-controller` is giving me following error, as if it is not using the kubelet identity (which has AcrPull role assigned):

`UNAUTHORIZED: authentication required, visit https://aka.ms/acr/authorization for more information.` when trying to GET Access Token from ACR. 

```apiVersion: image.toolkit.fluxcd.io/v1beta1
kind: ImageRepository
metadata:
  name: example-app
  namespace: flux-system
spec:
  image: test.azurecr.io/example-app/example-app
  interval: 1m0s
```
Have I missed something?

**Comment by danieltoth55** [comment]
@michalgozdowski 
I think the issue is that the current version of AKS Flux extension ([1.7.0](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/extensions-release#170-march-2023)) uses Flux v0.39.0, which only has v0.24.0 of the `image-reflector-controller`.
Version 0.25.0 of the controller introduced some [breaking changes](https://github.com/fluxcd/image-reflector-controller/blob/main/CHANGELOG.md#0250) which makes it possible to define `spec.provider` as `azure` in the `ImageRepository` definition. Without this setting the controller will use a generic approach for authentication (like username, password).
As a workaround you should be able to create a secret with username, password from your ACR and use that in your `ImageRepository` definition (`.spec.secretRef`).
Another approach would be to install Flux through Helm, not using the AKS extension...
